documentation/quarkus/guide-for-testing.asciidoc

```diff
@@ -0,0 +1,59 @@
+Quarkus provides multiple implementations of QuarkusTestResourceLifecycleManager for non-specific application requirements. For specific application requirements, it is advantageous to create your own implementations. If a custom QuarkusTestResourceLifecycleManager is created that needs to inject something into a test class, then inject methods can be used. It should be noted that these injections are performed only after the CDI has injected all necessary injections into the test class. These injections are not under the control of the CDI. 
+It is also possible to write your own test resources. These test resources are activated and configured by annotations. Here the @QuarkusTestResource annotation must be set to an annotation, which executes the two functions.
+
+7. detect hangs
+With the help of the annotation @QuarkusTest unexpected hangs can be diagnosed. This is done by outputting a stack trace on the console if no JUnit callbacks are called after a certain time. The timeout is 10 minutes by default. To be able to continue the test, this timeout can be controlled with quarkus.test.hang-detection-timeout. 
+
+8. testing native files
+In Quarkus it is possible to test native executable files. This is done with the annotation @NativeImageTest. However, in the future the functionality of the @NativeImageTest will be covered by the @QuarkusIntegrationTest. This will be used to launch as well as test the artifact generated by the Quarkus build. It also supports testing native images and container images as well as jar files. The @QuarkusIntegrationTest is a black box test just like the @NativeImageTest, which includes the same features and limitations. It is important to note that the @QuarkusIntegrationTest should be run as part of the integration suite.  The reason for this is that if the tests are run in the same phase as @QuarkusTest, they will not work because the final artifact has not yet been created.
+@QuarkusTests must not be mixed with other tests in a single execution run, such as a single Maven Surfire Plugin execution. This is because the @QuarkusTest, when running the entire execution flow, starts a Quarkus server and prevents other tests from starting their own Quarkus server. To work around this limitation the JUnit5 tag io.quarkus.test.junit.QuarkusTest is used.
+
+Source: https://quarkus.io/guides/getting-started-testing

```

Comment:
I think you can also show or mention the ability to use [continious testing](https://quarkus.io/guides/continuous-testing) with the developement mode in quarkus.